editor/src/components/canvas/ui-jsx-canvas-renderer/scene-component.tsx

```diff
@@ -27,4 +28,35 @@ export const SceneComponent = betterReactMemo(
       </Scene>
     )
   },
+  (
+    prevProps: React.PropsWithChildren<SceneProps>,
+    nextProps: React.PropsWithChildren<SceneProps>,
+  ) => {
+    // Compare child types (to see if a child was actually changed), and compare style
+    const prevChildren = React.Children.toArray(prevProps.children)
+    const nextChildren = React.Children.toArray(nextProps.children)
+
+    const childrenMatch =
+      prevChildren.length === nextChildren.length &&
+      prevChildren.every((child, index) => childUnchanged(child, nextChildren[index]))
+
+    return (
+      childrenMatch &&
+      fastDeepEquals(prevProps.style, nextProps.style) &&
+      fastDeepEquals(prevProps['data-label'], nextProps['data-label'])
+    )
+  },
 )
+
+type ReactChild = Exclude<React.ReactNode, boolean | null | undefined>
+
+function childUnchanged(prevChild: ReactChild, nextChild: ReactChild): boolean {
+  if (typeof prevChild === 'string' || typeof prevChild === 'number') {
+    return nextChild === prevChild
+  } else if (React.isValidElement(prevChild)) {
+    return React.isValidElement(nextChild) && prevChild.type === nextChild.type
+  } else {
+    // FIXME Fragments are all that is left
```

Comment:
We found that actually fragments aren't left over here, since they get squashed in the `React.Children.toArray` call above. However, we should add something here to explicitly capture that in case that changes, as well as any other exotic elements.